metabase/metabase
# fix sandboxing parameter field disabled

Closes https://github.com/metabase/metabase/issues/29143

### Description

Question parameters in the sandboxing modal used the old `SavedQuestionLoader` which loads a saved question first and then loads the question metadata. Even though loading card data went through redux, the code actually retrieved card data from the returned promise like `const card = await this.props.fetchQuestion(questionId);`. As you may know, the entity's `fetch` action tracks the request state in the store which prevents firing multiple requests for the same data. The reproduction steps render two fields, each of them called this action but the second invocation does not actually trigger a new request since there is an existing one in progress, hence the second action itself resolves with `undefined`. Overall, the approach of getting data from action is a hack and here it led to a bug. I replaced it with our Question loader which selects data from the store.

### How to verify

Run Metabase EE

1. Create a group, user, and sandbox per the repro steps in https://github.com/metabase/metabase/issues/29135
2. Refresh the permissions page and edit the sandbox you just created. Scroll down in the sandboxing modal.
3. Under "Parameter or variable", ensure both "State" and "Source" parameters selects are enabled

### Demo

<video src="https://user-images.githubusercontent.com/14301985/225758866-8d02ab2c-b965-47b4-84da-8fe72f46f84a.mov" />

### Checklist



- - -
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/metabase/metabase/29256)
